canonical-web-and-design/maas-ui
# feat(ui): allow either LXD or libvirt when deploying machine as VM host

## Done

- Added radio buttons to the deploy form to allow either LXD or libvirt VM hosts

## QA

### MAAS deployment

To run this branch you will need access to one of the following MAAS deployments:

- [Karura](/HACKING.md#karura)
- [Bolla](/HACKING.md#bolla)
- [A development MAAS](/HACKING.md#development-deployment)
- [A local snap MAAS](/HACKING.md#snap-deployment) (this will not usually have machines)

### Running the branch

You can run this branch by:

- Serving with [dotrun](/HACKING.md#maas-ui-development-setup)
- [Building in a development MAAS](/HACKING.md#running-maas-ui-from-a-development-maas)

### QA steps

- Open the network tab in the browser console check that the websocket connection can be found
- Open the Deploy form for a machine and submit the form without checking the "Deploy KVM" checkbox
- Check that the websocket request does not include either `install_kvm` or `register_vmhost`
- Cancel deploying and open again, this time checking the "Deploy KVM" checkbox and the "LXD" radio button before submitting
- Check that the websocket request includes `register_vmhost` but not `install_kvm`
- Cancel deploying and open again, this time checking "libvirt" before submitting
- Check that the websocket request includes `install_kvm` but not `register_vmhost`

## Fixes

Fixes #2469 

## Screenshot

![Screenshot_2021-04-12 Machines bolla MAAS](https://user-images.githubusercontent.com/25733845/114328579-69517f80-9b80-11eb-852a-16443f092110.png)
